### ISSUE TITLE
Make API for lazy-loaded assemblies consistent across project file and LoadAssembly

### ISSUE THREAD

**Issue by captainsafia** [dotnet/aspnetcore#24814]
During today's community standup, a user brought up the inconsistency in how DLLs are marked in the project file (without the extension) and loaded in the code (with the extension).

We should make these consistent. ~~This will likely mean removing the extension from the load API.~~

**Comment by campersau** [comment]
Not using the ".dll" extension might cause some confusion because often you would lazy load external libraries which are included by a `PackageReference` and it would **often** be the same string as the one in `BlazorWebAssemblyLazyLoad`.

So some might mistakenly assume that transitive NuGet dependencies are also lazy loaded automatically (escpecially for meta packages) because `PackageReference` does.

And when the DLL names inside of NuGet packages don't match the NuGet package id (e.g. [`Blazor-Redux`](https://www.nuget.org/packages/Blazor-Redux/) vs `BlazorRedux.dll`) it might be confusing.

```xml
<ItemGroup>
  <PackageReference Include"Blazor-Redux" Version="0.0.6" />
</ItemGroup>
<ItemGroup>
  <BlazorWebAssemblyLazyLoad Include="BlazorRedux" />
  <!-- vs -->
  <BlazorWebAssemblyLazyLoad Include="BlazorRedux.dll" />
</ItemGroup>
```

---

Since this is a very technical feature and if this behavior is properly documented it might not matter much but just wanted to share my thoughts about it.

**Comment by adrianwright109** [comment]
@captainsafia - I would like to take this issue and submit a PR - any objections ?

**Comment by captainsafia** [comment]
@adrianwright109 No objections! For this issue, I'm in agreement with @campersau that the project file attribute should take the filename with the DLL to make it explicit that assemblies do not equal NuGet package names.

You'll need to make the modification [here](https://github.com/dotnet/aspnetcore/blob/4092201629a7feea1ead23202aff4ee8f5b77664/src/Components/WebAssembly/Sdk/src/GenerateBlazorWebAssemblyBootJson.cs). Let me know if you have any questions.

**Comment by captainsafia** [comment]
Resolved in #24866.
